core/domain/activity_jobs_one_off.py

```diff
@@ -589,3 +589,154 @@ def map(snapshot_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class ValidateSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that validates the commit logs of the SnapshotMetadataModels.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        # Considering all the models that has a SnapshotMetadata except
+        # config model because it does not have commit logs.
+        return [
+            collection_models.CollectionSnapshotMetadataModel,
+            collection_models.CollectionRightsSnapshotMetadataModel,
+            exp_models.ExplorationSnapshotMetadataModel,
+            exp_models.ExplorationRightsSnapshotMetadataModel,
+            question_models.QuestionSnapshotMetadataModel,
+            skill_models.SkillSnapshotMetadataModel,
+            story_models.StorySnapshotMetadataModel,
+            topic_models.TopicSnapshotMetadataModel,
+            topic_models.SubtopicPageSnapshotMetadataModel,
+            topic_models.TopicRightsSnapshotMetadataModel
+        ]
+
+    @staticmethod
+    def map(snapshot_model):
+
+        class_name = snapshot_model.__class__.__name__
+        missing_commit_log_msg = 'MISSING COMMIT LOGS-'
+
+        if isinstance(
+                snapshot_model,
+                topic_models.SubtopicPageSnapshotMetadataModel):
+            # The subtopic snapshot ID is in the format
+            # '<topicId>-<subtopicNum>-<version>'.
+            topic_id, subtopic_num, version = snapshot_model.id.split('-')
+            model_id = '%s-%s' % (topic_id, subtopic_num)
+        else:
+            model_id, version = snapshot_model.id.split('-')
+
+        if isinstance(
```

Comment:
Try to find a way to write this more declaratively / reduce the branching. E.g. you could make a dict mapping model names to the associated subclasses and then do a for loop on that dict. (Maybe it makes sense to force each model class to define a get_commit_log_id(model_id, version) function too.)

Reply:
Done.